### DISCUSSION TITLE
shrinking vs. clipping

### DISCUSSION THREAD

**Discussion by pr-apes** [speedata/publisher#400] [Q&A] [answered]
@pgundlach,

I’’m testing the new clipping feature. It is really great (many thanks for the implementation).

But I’m afraid I don’t get what the difference between shrinking and clipping is.

Here you have the following sample (to be run with `sp --dummy`):

``` xml
<Layout xmlns="urn:speedata.de:2009/publisher/en"
  xmlns:sd="urn:speedata:2009/publisher/functions/en">
  <Pageformat height="14cm" width="11cm" />

	<Trace grid="yes" gridallocation="yes"/>
	
  <Record element="data">
    <PlaceObject>
      <Clip left="1cm" right="2cm" top="1cm" bottom="2cm" method="shrink">
        <Image width="5cm" file="_sampleb.pdf" />
      </Clip>
    </PlaceObject>
    <PlaceObject column="5" >
	  <Clip left="1cm" right="2cm" top="1cm" bottom="2cm" method="clip">
        <Image width="5cm" file="_sampleb.pdf" />
      </Clip>
    </PlaceObject>
	<ClearPage/>
    <PlaceObject>
      <Clip left="1cm" width="1cm" top="1cm" height="1cm" method="shrink">
        <Image width="5cm" file="_sampleb.pdf" />
      </Clip>
    </PlaceObject>
    <PlaceObject column="5" >
	  <Clip left="1cm" width="1cm" top="1cm" height="1cm" method="clip">
        <Image width="5cm" file="_sampleb.pdf" />
      </Clip>
    </PlaceObject>
  </Record>
</Layout>
```

Other than placement. I cannot see any size difference between both images (no matter which method is used).

What am I missing here? Or, which is the difference in output size between both methods?

Many thanks for your help.

**Comment by pgundlach** [comment, marked as answer]
`method="clip"` keeps the original object but hides everything not selected by the left/right/width/height/... attributes. Does this help? You probably don't need `method="clip"`

**Reply by pr-apes** [reply]
I cannot pretend I understand the difference, but you explanation confirmed my suspicions (`method=shrink` is what I really need).

Now I got it (I just saw the light 😅): 

> If _Publisher_ had scissors, shrinking would cut the image itself, while clipping would cut a frame to be placed on top of the image (enabling partial display).

Many thanks for your help.
